pytripgui/canvas_vc/canvas_controller.py

```diff
@@ -90,10 +92,38 @@ def set_patient(self, patient, state):
             self._ui.voi_list.event_callback = self._on_update_voi
             self._ui.voi_list.fill(patient.vdx.vois, lambda item: item.name)
 
-        self._ui.set_position_changed_callback(self.set_current_slice_no)
         self.update_canvas_view()
+
+        self._update_slider_values(position_safe=True, perspective_safe=True)
+
         self._ui.draw()
 
+    def _update_slider_values(self, position_safe: bool, perspective_safe: bool):
+        """
+        Updates min, max and current values of ui slider.
+        Can be "safe" position-callback-wise and perspective-callback-wise,
+            which means that callback will not be invoked id proper flag is set
+
+        :param bool position_safe: if set, position change callback will be suppressed
+        :param bool perspective_safe: if set, perspective change callback will be suppressed
+        """
+        # remove event listeners
+        if position_safe:
+            self._ui.remove_position_changed_callback(self.set_current_slice_no)
+        if perspective_safe:
+            self._ui.internal_events.on_perspective_change -= self._perspective_has_changed_callback
+
+        # set parameters
+        self._ui.max_position = self._model.projection_selector.last_slice_no
+        self._ui.position = self._model.projection_selector.current_slice_no
+        self._ui.perspective = self._model.projection_selector.plane
+
+        # add event listeners back
+        if position_safe:
+            self._ui.set_position_changed_callback(self.set_current_slice_no)
+        if perspective_safe:
+            self._ui.internal_events.on_perspective_change += self._perspective_has_changed_callback
```

Comment:
Why `+=` is used here ?

Reply:
It is the way listenres are added - I just copied that line from contructor

Reply:
ok